### PR TITLE
DAOS-5061 tests: Set memlock ulimit values to unlimited

### PR DESCRIPTION
**Pull request by phender**
Set memlock ulimit value to unlimited to avoid cannot allocate memory
errors with verbs.

Test-tag: pr daily_regression

Signed-off-by: Phillip Henderson <phillip.henderson@intel.com>

### CODE REVIEW

**Review by mchaarawi — COMMENTED** [review]
it would be great to print output of
ulimit -a
on all the nodes where HW tests are running for info/verification of the ulimit values

**Comment by phender** [comment]
Test failures in https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-7328/1/ are known issues:

- Test Hardware / Functional Hardware Small / FTEST_control.DmgPoolQueryTest.1-./control/dmg_pool_query_test.py:DmgPoolQueryTest.test_pool_query_basic;container-nvme-rebuild-scm-hosts-ior-pool-pool_uuids-servers-6e5e
  - https://daosio.atlassian.net/browse/DAOS-9030
- Test Hardware / Functional Hardware Large / FTEST_pool.PoolCreateTests.1-./pool/create_capacity.py:PoolCreateTests.test_create_pool_quantity;hosts-pool-0-timeouts-3d5b
  - https://daosio.atlassian.net/browse/DAOS-8996

File: src/tests/ftest/scripts/setup_nodes.sh [hunk 1]
```diff
@@ -42,6 +42,10 @@ sudo bash -c "set -ex
 if [ \"\$(ulimit -c)\" != \"unlimited\" ]; then
     echo \"*  soft  core  unlimited\" >> /etc/security/limits.conf
 fi
+if [ \"\$(ulimit -l)\" != \"unlimited\" ]; then
+    echo \"*  soft  memlock  unlimited\" >> /etc/security/limits.conf
+    echo \"*  hard  memlock  unlimited\" >> /etc/security/limits.conf
+fi
```
Comment (brianjmurrell) [inline src/tests/ftest/scripts/setup_nodes.sh:48]: To be perfectly clear (and probably preach to the choir), this only affects our local testing.

I left a question open in the ticket about how production systems become compliant.

**Review by brianjmurrell — APPROVED** [review]

File: src/tests/ftest/scripts/setup_nodes.sh (hunk 1 above)
Comment (phender) [reply src/tests/ftest/scripts/setup_nodes.sh:62]: Who should answer this question? @johannlombardi

**Comment by phender** [comment]
> it would be great to print output of ulimit -a on all the nodes where HW tests are running for info/verification of the ulimit values

This has been added.  Example: https://build.hpdd.intel.com/blue/rest/organizations/jenkins/pipelines/daos-stack/pipelines/daos/branches/PR-7328/runs/3/nodes/1207/log/?start=0

```
[2021-11-18T13:18:30.611Z] + ulimit -a
[2021-11-18T13:18:30.611Z] core file size          (blocks, -c) 0
[2021-11-18T13:18:30.611Z] data seg size           (kbytes, -d) unlimited
[2021-11-18T13:18:30.611Z] scheduling priority             (-e) 0
[2021-11-18T13:18:30.611Z] file size               (blocks, -f) unlimited
[2021-11-18T13:18:30.611Z] pending signals                 (-i) 761517
[2021-11-18T13:18:30.611Z] max locked memory       (kbytes, -l) 2097152
[2021-11-18T13:18:30.611Z] max memory size         (kbytes, -m) unlimited
[2021-11-18T13:18:30.611Z] open files                      (-n) 1024
[2021-11-18T13:18:30.611Z] pipe size            (512 bytes, -p) 8
[2021-11-18T13:18:30.611Z] POSIX message queues     (bytes, -q) 819200
[2021-11-18T13:18:30.611Z] real-time priority              (-r) 0
[2021-11-18T13:18:30.611Z] stack size              (kbytes, -s) 8192
[2021-11-18T13:18:30.611Z] cpu time               (seconds, -t) unlimited
[2021-11-18T13:18:30.611Z] max user processes              (-u) 761517
[2021-11-18T13:18:30.611Z] virtual memory          (kbytes, -v) unlimited
[2021-11-18T13:18:30.611Z] file locks                      (-x) unlimited
```

**Review by brianjmurrell — APPROVED** [review]

File: src/tests/ftest/scripts/setup_nodes.sh (hunk 1 above)
Comment (brianjmurrell) [reply src/tests/ftest/scripts/setup_nodes.sh:62]: I'm a little concerned that https://daosio.atlassian.net/browse/DAOS-5061?focusedCommentId=89879 means that this limits change is only needed on clients.  We are applying it to all nodes, including servers.  This has the potential to mask a bug where the server node is not being modified correctly by daos.  That is, everything will work in our CI but will fail in a production (i.e. end-user) environment.

I wonder if making this change needs to be pushed out into the test framework to only be done on clients.

**Review by jolivier23 — APPROVED** [review]

**Comment by phender** [comment]
2 of the 5 failures in https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-7328/8/ are known issues:
- HW Medium / 1-./osa/online_reintegration.py:OSAOnlineReintegration.test_osa_online_reintegration
  - https://daosio.atlassian.net/browse/DAOS-9176
- HW Medium / 3-./osa/online_reintegration.py:OSAOnlineReintegration.test_osa_online_reintegration_without_csum
  - https://daosio.atlassian.net/browse/DAOS-9176

3 of the 5 failures in https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-7328/8/ are new issues unrelated to the change:
- HW Medium / 1-./osa/offline_drain.py:OSAOfflineDrain.test_osa_offline_drain
  - https://daosio.atlassian.net/browse/DAOS-9203
- HW Medium / 2-./osa/offline_drain.py:OSAOfflineDrain.test_osa_offline_drain_during_aggregation
  - https://daosio.atlassian.net/browse/DAOS-9203
- HW Medium / 3-./osa/offline_drain.py:OSAOfflineDrain.test_osa_offline_drain_after_snapsot
  - https://daosio.atlassian.net/browse/DAOS-9203

**Pull request merged**